src/SonarQube.Client/Api/V7_20/GetIssuesWithComponentRequest.cs

```diff
@@ -0,0 +1,56 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Newtonsoft.Json;
+
+namespace SonarQube.Client.Api.V7_20
+{
+    internal class GetIssuesWithComponentRequest : GetIssuesRequest
+    {
+        // this class should not be used on it's own
+        private protected GetIssuesWithComponentRequest(){}
```

Comment:
Is there a strong reason to have `private protected` instead of `protected` here? The class is already `internal`, so the `private` doesn't really do anything

Reply:
Not really, since as you said it doesn't do anything. Ideally I would love to make it explicit that this class shouldn't be inherited from without a good reason, but this doesn't really achieve that